streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -384,18 +384,20 @@ public void foreach(final ForeachAction<? super K, ? super V> action) {
 
     @Override
     public KStream<K, V> through(final String topic) {
-        return through(topic, Produced.with(null, null, null));
+        return through(topic, Produced.with(keySerde, valSerde, null));
```

Comment:
Is this ok to do? I was surprised while writing my test that the serdes didn't get propagated all the way to the sink. (question applies to all changes in this file)

Reply:
Good call, I think it is okay.